app/javascript/components/bootcamp/SolveExercisePage/TestResultsView/InspectedTestResultView.tsx

```diff
@@ -9,9 +9,17 @@ import {
 import { TestResultInfo } from './TestResultInfo'
 import { PassMessage } from './PassMessage'
 import useTestStore from '../store/testStore'
+import useTaskStore from '../store/taskStore/taskStore'
 
 function _InspectedTestResultView() {
   const { result, viewContainerRef, firstExpect } = useInspectedTestResultView()
+  const { testSuiteResult } = useTestStore()
+  const { wasFinishLessonModalShown } = useTaskStore()
+
+  const isSpotlightActive = useMemo(() => {
+    if (!testSuiteResult) return false
+    return !wasFinishLessonModalShown && testSuiteResult.status === 'pass'
```

Comment:
I have a preference for not having && where possible and also not having `!x && y`. I think it's clearer to have an extra guard clause.

```suggestion
    if (!testSuiteResult) return false
    if (wasFinishLessonModalShown) return false
    
    return testSuiteResult.status === 'pass'
```